contracts/src/LightClientV2.sol

```diff
@@ -82,31 +89,45 @@ contract LightClientV2 is LightClient {
         }
         // format validity check
         BN254.validateScalarField(newState.blockCommRoot);
+        BN254.validateScalarField(nextStakeTable.blsKeyComm);
+        BN254.validateScalarField(nextStakeTable.schnorrKeyComm);
+        BN254.validateScalarField(nextStakeTable.amountComm);
 
         // epoch-related checks
         uint64 lastUpdateEpoch = currentEpoch();
-        uint64 newEpoch = epochFromBlockNumber(newState.blockHeight, _blocksPerEpoch);
-        // advancing 1 epoch is only allowed if the epoch root (last block - 5) of the last epoch
-        // was submitted
-        // or if there is no last epoch (i.e. when current epoch is epoch 1)
-        require(newEpoch < lastUpdateEpoch + 2, MissingEpochRootUpdate());
-        if (
-            lastUpdateEpoch != 0
-                && epochFromBlockNumber(newState.blockHeight + 5, _blocksPerEpoch) != lastUpdateEpoch
-        ) {
-            require(isEpochRoot(finalizedState.blockHeight), MissingEpochRootUpdate());
+        uint64 newEpoch = epochFromBlockNumber(newState.blockHeight, blocksPerEpoch);
```

Comment:
`newEpoch` is a bit of a confusing name here. `newStateEpoch`?

Reply:
okay, happy to change.

Reply:
done in [9e7e6ea](https://github.com/EspressoSystems/espresso-network/pull/2984/commits/9e7e6eacfbee5a253d6ed53b525df62d6c7d5006)